todoapp/users/models.py

```
@@ -4,13 +4,40 @@
 from django.contrib.auth.models import BaseUserManager
 from django.contrib.auth.models import PermissionsMixin
 from django.db import models
-
-
+from django.utils.translation import gettext_lazy as _
+ 
 class UserManager(BaseUserManager):
     """
     Add manager methods here to create user and super user
     """
 
+    def create_user(self, email, password, **extra_fields):
+        """
+        Create and save a user with the given email and password.
+        """
+        if not email:
```

What is password is null?

---

It will call set_password(None). this will not immediately raise an error but store it as unusable password which means the user cannot authenticate with it. we can raise an exception for it.